apps/dashboard/app/models/announcement.rb

```diff
@@ -31,10 +32,36 @@ def msg
     msg.blank? ? nil : msg
   end
 
+  def button_text
+    opts.fetch(:button_text, I18n.t('dashboard.announcements_button_text')).to_s
+  end
+
   # Whether this is a valid announcement
   # @return [Boolean] whether it is valid
   def valid?
-    !!msg
+    return false unless msg
+
+    return false if dismissible? && !id
+
+    true
+  end
+
+  def completed?
+    dismissible? && user_settings.dig(:announcements, id.to_sym)
+  end
+
+  def dismissible?
+    required? || opts.fetch(:dismissible, false)
+  end
+
+  def required?
+    opts.fetch(:required, false)
+  end
+
+  def id
+    default_id = @path.basename.to_s if @path
+    # default_id = Digest::SHA1.hexdigest(@path.basename.to_s) if @path
+    opts.fetch(:id, default_id)
```

Comment:
OSC reuses names all the time. We publish the `ondemand_update.yml` all the time and it has a time check within it to show the alert based on the current time.

In any case - I feel like the id should be a hexdigest of the content of the file so folks can continue to reuse the same filename and not have ot say publish _1.yml and _2.yml and so on forever.

Circling back to my feeling that all announcements should be dismissable - by making the id the content of the file we can avoid collisions and having to increment or put dates in filenames.,

Reply:
I have updated to use the `hexdigest` of the message.